util/dvsim/NcLauncher.py

```diff
@@ -150,14 +160,49 @@ def poll(self):
         """
 
         assert self.process is not None
-        elapsed_time = datetime.datetime.now() - self.start_time
-        job_runtime_secs = elapsed_time.total_seconds()
         if self.process.poll() is None:
-            timeout_mins = self.deploy.get_timeout_mins()
-            if timeout_mins is not None and not self.deploy.gui:
-                if job_runtime_secs > (timeout_mins * 60):
+            # get the run_timeout value from the test hjson or dvsim switch
+            # define a wait_timeout to 3 hours (hardcoded in this file)
+            # define a file size threshold on the run log for the algo to
+            # predict the state of the job (running or waiting for resource)
+            run_timeout_mins = self.deploy.get_timeout_mins()
+            wait_timeout_mins = 180  # max wait time in job / license queue
+            file_size_thresh_bytes = 5120  # log file size threshold
+            if run_timeout_mins is not None and not self.deploy.gui:
+                # query the log file size
+                f_size = os.path.getsize(self.deploy.get_log_path())
+
+                if f_size >= file_size_thresh_bytes:
+                    if (self.nc_job_state == 'waiting'):
+                        # if the job log size is more than the threshold, start the
+                        # the job run time counter if has not already been started.
+                        self.start_time = datetime.datetime.now()
+                        self.nc_job_state = 'running'
+
+                if self.nc_job_state == 'waiting':
+                    # if the job log size is less than the threshold, increment
```

Comment:
I really like the structure of the if cases here! But I think this comment could be made slightly clearer. Something like this maybe? "If we get here, we know the log size is less than the threshold. Update the job wait time counter."

(And "increment" is probably the wrong verb: it's not like we've got `elapsed_wait_time += ...`)

Reply:
done . added clarity to the comment